### PR TITLE
[WC-14] Implement Home Page

### PR DESCRIPTION
**Pull request by tong788**
This pull request is about implementation of homepage consisting of components such as navbar, course, and footer. In this page, I have provided the link to login and sign-up page too.  

[//]: # 'Please include a summary of the changes and the related issue. (You can write below without deleting this comment)'

### What

[//]: # 'Briefly describe this feature / fix / refactor (You can write below without deleting this comment)'

### Why

[//]: # 'Describe the reason why we need this feature / fix / refactor or refer to Jira Ticket (You can write below without deleting this comment)'

## Type of change

Please check the relevent option with `x`.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if needed)
- [ ] My changes generate no new warnings
- [ ] I have tested on my `Local Machine`

## Release Notes

- **New Features**
  - Introduced multiple new components: `ClientReview`, `Contact_us`, `Footer`, `ForgetPassword`, `Login`, `MostPopularCourse`, `RecentCourses`, and `RecommendedForYou`.
  - Enhanced the home page layout with new sections for popular courses, recent courses, recommendations, client reviews, and a contact section.
  - Added a password recovery feature with a user-friendly interface.

- **Bug Fixes**
  - Added custom spelling exception for "fname" in code spell checker settings.

- **Documentation**
  - Updated data files to include course details for client reviews, recently viewed courses, and recommended courses.

### CODE REVIEW

**Comment by Billy19191** [comment]
Good good, also don't forget to add what and why. GL ⭐

File: app/client/login/page.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import Login from '../components/Login'
+
+export default function login() {
+  return (
+    <>
+      <Login />
+    </>
+  )
+}
```
[the remark this answers is not in the document]
Comment (tong788) [reply app/client/login/page.tsx:10]: Can you explain about suspense and error boundary in React?

[the remark this answers is not in the document]
Comment (tong788) [reply app/client/login/page.tsx:10]: what is fallback UI?

[the remark this answers is not in the document]
Comment (tong788) [reply app/client/login/page.tsx:10]: Ok I fixed it already

File: app/client/login/forget-password/page.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import ForgetPassword from '../../components/ForgetPassword'
+import React from 'react'
+
+export default function forgetpassword() {
+  return (
+    <>
+      <ForgetPassword />
+    </>
+  )
+}
```
[the remark this answers is not in the document]
Comment (tong788) [reply app/client/login/forget-password/page.tsx:10]: There will be conflict with other component if I changed.

[the remark this answers is not in the document]
Comment (tong788) [reply app/client/login/forget-password/page.tsx:10]: I will fix it later.

**Pull request merged**